Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -136,6 +167,47 @@ noncomputable def lanCompColimIso (L : C ⥤ D) [∀ (G : C ⥤ H), L.HasLeftKan
       congr 1
       exact congr_app (L.lanUnit.naturality f) i))
 
+variable [HasColimitsOfShape D H]
+
+variable (G : C ⥤ H) [L.HasPointwiseLeftKanExtension G]
+
+omit [∀ (F : C ⥤ H), L.HasLeftKanExtension F]
+
+@[local instance]
+private theorem colimitIsoColimitGrothendieck_inst₁ :
+    HasColimit (CostructuredArrow.grothendieckProj L ⋙ G) :=
+  hasColimit_of_hasColimit_fiberwiseColimit_of_hasColimit _
+
+variable [HasColimitsOfShape C H]
+
+/-- If `G : C ⥤ H` is a left Kan extension of a functor `L : C ⥤ D` and `H` has colimits of shape
+`C`, `D`, and `CostructuredArrow L X` for all `X : D`, then the colimit of `G` is isomorphic to
+the colimit of a canonical functor `Grothendieck (CostructuredArrow.functor L) ⥤ H` induced by
+`L` and `G`. -/
+@[simps!]
+noncomputable def colimitIsoColimitGrothendieck :
+    colimit G ≅ colimit (CostructuredArrow.grothendieckProj L ⋙ G) :=
+  ((lanCompColimIso L).app G).symm ≪≫
+  HasColimit.isoOfNatIso (lanObjIsoFiberwiseColimit L G) ≪≫
+  colimitFiberwiseColimitIso _
```

Comment:
Optional suggestion: as someone who didn't already know why the result is true, I find this easier to read:
```suggestion
    colimit G ≅ colimit (CostructuredArrow.grothendieckProj L ⋙ G) := calc
  colimit G
    ≅ (L.lan ⋙ colim).obj G := ((lanCompColimIso L).app G).symm
  _ ≅ colimit (fiberwiseColimit (CostructuredArrow.grothendieckProj L ⋙ G)) :=
        HasColimit.isoOfNatIso (lanObjIsoFiberwiseColimit L G)
  _ ≅ colimit (CostructuredArrow.grothendieckProj L ⋙ G) :=
        colimitFiberwiseColimitIso _
```